### PR TITLE
Update modules.md

### PR DESCRIPTION
**Pull request by ean5533**
* Add some missing descriptions for notable modules like `misk` and `misk-api`
* Flesh out the intro a bit with general module strategy info

### CODE REVIEW

File: docs/modules.md [hunk 1]
```diff
@@ -89,3 +111,12 @@ implementation.
 ## misk-events
 
 An event publisher + consumer. There is no open source reference implementation at this time.
+
+
+## wisp.*
+
+These modules were created to extract specific pieces of functionality out of the `misk*` modules
+into new, low-dependency modules. They were especially focused on having no Guice dependencies.
+Some of these modules duplicate existing Misk functionality.
```
Comment (adrw) [inline docs/modules.md:120]:
```suggestion
Some of these modules duplicate existing Misk functionality, but over time implementations will be deduplicated as part of broader code cleanup efforts.
```

File: docs/modules.md [hunk 2]
```diff
@@ -89,3 +111,12 @@ implementation.
...
+
+Now, a few of these modules are starting to be folded back into `misk*`, or at least deduplicated.
```
Comment (adrw) [inline docs/modules.md:122]:
```suggestion
```

This might be a bit forward for public docs.

File: docs/modules.md [hunk 3]
```diff
@@ -12,6 +24,11 @@ request object and return a response object. Throw an exception like `BadRequest
 fail the request without much boilerplate.
 
 
+## misk-api
+
+High level interfaces and data classes. This module is agnostic to implementation details.
```
Comment (adrw) [inline docs/modules.md:29]:
```suggestion
High level interfaces and data classes which are implemented by both misk and wisp modules. This module is agnostic to implementation details.
```

File: docs/modules.md [hunk 4]
```diff
@@ -1,6 +1,18 @@
...
+
+In general, our module strategy is **TODO**
+
+We use testfixtures **TODO*
+
+
+## misk
+
+Most of the implementation of Misk's web server components.
+There is a lot of stuff in this module. Pieces are occasionally extracted into new modules.
```
Comment (adrw) [inline docs/modules.md:13]:
```suggestion
This is the original module, more and more pieces are being extracted into new modules for improved organization, faster builds, and smaller dependencies for downstream customers.
```

**Review by adrw — APPROVED** [review]

**Review by frojasg — APPROVED** [review]
LGTM. Thanks for improving the docs!

**Comment by ean5533** [comment]
* Add some missing descriptions for notable modules like misk and misk-api
* Flesh out the intro a bit with general module strategy info

**Pull request merged**